api/src/main/java/org/openmrs/module/dhisconnector/api/impl/DHISConnectorServiceImpl.java

```diff
@@ -806,7 +806,15 @@ public String uploadMappings(MultipartFile mapping) {
 		String mappingFolderName = OpenmrsUtil.getApplicationDataDirectory() + DHISCONNECTOR_MAPPINGS_FOLDER
 		        + File.separator;
 		String mappingName = mapping.getOriginalFilename();
-		
+
+		File mappingsFolder = new File(mappingFolderName);
+		mappingsFolder.mkdirs();
+
+		if (!mappingsFolder.isDirectory()) {
+			msg = Context.getMessageSourceService().getMessage("dhisconnector.uploadMapping.directoryNotCreated");
+			return msg;
```

Comment:
Shall we remove this redundant variable?

Reply:
Sure

Reply:
Resolved with the commit fbd478d
@jayasanka-sack please review